department-of-veterans-affairs/abd-vro
# MCP-1948: Sinusitis processor

# Sinusitis processor 

## Description



### What was the problem?
No support for sinusitis condition (`6510`).
CFR https://www.law.cornell.edu/cfr/text/38/4.97



### How does this fix it?
Adds python service for sinusitis condition logic. Includes Medication, Procedure and Condition filtering and SME validated codesets. 

Calculated evidence summary fields:
`constantSinusitis` is true only when a Sinusitis diagnosis 'onsetDate' is more than 12 weeks prior to the date of claim.
`radicalSurgery` is true only when a procedure matches the CPT procedures identified as radical surgery. 
`multipleSurgery` is true only when more than two relevant surgeries have been identified. 

The calculated evidence summary is intended to make clear the eligibility of a claim for the fast track. The summary will further the ability of the VRO user to identify claims with sufficient evidence for (most likely a positive) rating. If any of the calculated evidence summary fields are true than the claim should be routed on the fast track. 

There are testing limitations for the procedures algorithm because the sandbox does not supply codes for procedurs and text matching is not optimal for the CPT code system. 



### Jira Tickets



- [MCP-000](https://amida.atlassian.net/browse/MCP-000)

## How to test this PR





    Step 1
    ./gradlew -PenablePrototype=true build check docker:
    -export COMPOSE_PROFILES=prototype

    ./gradlew :app:dockerComposeDown :app:dcPrune :app:dockerComposeUp

- go to Swagger page

- enter "6510" for diagnosticCode

ICN with partial evidence: 1000003, 1000005, 10000271, 17000151, 173, 25

ICN that qualifies for constant sinusitis: 195


## Reminders

- Review [Pull-Requests](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests) guidelines
[Roadmap](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Roadmap) wiki pages
